monkey/infection_monkey/island_api_client/i_island_api_client.py

```diff
@@ -54,3 +55,20 @@ def get_pba_file(self, filename: str) -> bytes:
         :raises IslandAPIError: If an unexpected error occurs while attempting to retrieve the
                                 custom PBA file
         """
+
+    @abstractmethod
+    def get_agent_binary(self, os_name: str) -> Optional[bytes]:
```

Comment:
Accept `OperatingSystem` instead of `str`. Let the `IIslandAPIClient` worry about how that should be represented (serialized) to the Island.